### PR TITLE
Support multiple inventory files

### LINKED ISSUE #975: Rebuild index job failed
First failed on yesterday's [scheduled run](https://github.com/nextstrain/nextstrain.org/actions/runs/10313465649):

> inventory for core - manifest key: nextstrain-data/config-v1/2024-08-08T01-00Z/manifest.json
> There was an error while fetching the S3 inventory for core. This is fatal.
>  The manifest file for the S3 inventory for bucket nextstrain-data includes more than one inventory file. This situation was not encountered during development, but this is presumably caused by the inventory size exceeding some threshold and being chunked into multiple files. Please check this is indeed the case and, if so, amend the code to parse and join each file.

I noticed when pushing ceb01dc3740f89ca9f6b47db63f069485b0969fd.

## Progress

- [x] #976
- [x] #979
- [ ] #981

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

This is necessary for the core collection inventory which is now split into 2 files.

## Related issue(s)

- #975

## Checklist

- [x] Checks pass
- [x] Check if changes affect the [resource index JSON revision](https://docs.nextstrain.org/projects/nextstrain-dot-org/en/latest/resource-collection.html#resource-index-revisions)
- [x] Local runs are successful ([ref](https://github.com/nextstrain/nextstrain.org/pull/979#issuecomment-2278879109))
- [x] Rebuild index job is successful ([ref](https://github.com/nextstrain/nextstrain.org/actions/runs/10327086205/job/28591692144))

### CODE REVIEW

**Comment by victorlin** [comment]
Local testing:

```
$ node ./resourceIndexer/main.js --collections core --save-inventories --output resources-from-remote.json.gz
inventory for core - manifest key: nextstrain-data/config-v1/2024-08-09T01-00Z/manifest.json
inventory for core - parsed manifest JSON
Saving manifest to ./devData/core.manifest.json
Saving inventory chunk to ./devData/core-bb94a413-28e5-4be4-94e7-529093545f3b.csv.gz
Saving inventory chunk to ./devData/core-f3db09e2-cb6b-4a47-a649-7453eadb5a65.csv.gz
inventory for core - fetched 542942 rows

$ node ./resourceIndexer/main.js --collections core --local --output resources-from-local.json.gz
inventory for core -- reading manifest from ./devData/core.manifest.json
inventory for core -- reading S3 inventory ./devData/core-bb94a413-28e5-4be4-94e7-529093545f3b.csv.gz
inventory for core -- reading S3 inventory ./devData/core-f3db09e2-cb6b-4a47-a649-7453eadb5a65.csv.gz
inventory for core - read 542942 rows from the local file

$ du -h resources*
 30M    resources-from-local.json.gz
 30M    resources-from-remote.json.gz

$ diff resources-from-remote.json.gz resources-from-local.json.gz
# same
```

**Comment by joverlee521** [comment]
Test app's `/pathogen` page doesn't work because the `v3.json` doesn't exist yet. 
Manually triggered the [index resources GH Action](https://github.com/nextstrain/nextstrain.org/actions/runs/10327086205).

**Comment by joverlee521** [comment]
Errr, there might be a bug in the GH Action workflow or scripts because the [manual run ended up uploading to as `v2.json`](https://github.com/nextstrain/nextstrain.org/actions/runs/10327086205/job/28591692144#step:7:30).

**Review by jameshadfield — APPROVED** [review]
I was wondering when this time would arrive! Thanks for the fix @victorlin

File: resourceIndexer/inventory.js [hunk 1]
```diff
@@ -271,14 +273,21 @@ function _parseManifest(manifest) {
...
-    inventory: `./devData/${name}.inventory.csv.gz`,
-  }
+function _localManifestPath(name) {
+  return `./devData/${name}.manifest.json`;
+}
+
+/**
+ * Returns the local filepath for the inventory associated with this collection
+ * @param {string} name Collection name
+ * @param {string} key Inventory key from the manifest
```
[the remark this answers is not in the document]
Comment (victorlin) [reply resourceIndexer/inventory.js:307]: Are you referring to the hyphens? Those are optional (quoted from the same doc page):

> If you provide a description, you can make the JSDoc comment more readable by inserting a hyphen before the description. Be sure to include a space before and after the hyphen.

I prefer the hyphens but I was just maintaining consistency with the existing contents. I can make a separate PR to add those if that's what you are referring to.

File: resourceIndexer/inventory.js [hunk 2]
```diff
@@ -51,6 +51,12 @@ const fetchInventoryRemote = async ({bucket, prefix, name, save}) => {
 
   console.log(`inventory for ${name} - parsed manifest JSON`)
 
+  const manifestPath = _localManifestPath(name);
+  if (save) {
```
[the remark this answers is not in the document]
Comment (victorlin) [reply resourceIndexer/inventory.js:55]: I believe this only impacts subsequent usage with `--local`. I've added some extra error handling in that case: 4322955db7056086ea821c45e4a9058f78bd5664

**Comment by victorlin** [comment]
@joverlee521 thanks for looking into the version stuff!

File: resourceIndexer/inventory.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jameshadfield) [reply resourceIndexer/inventory.js:307]: No - you have an argument with a single destructuring parameter, so the docstring should be something like 

```js
/**
 * Returns the local filepath for the inventory associated with this collection
 * @param {Object} param
 * @param {string} param.name Collection name
 * @param {string} param.key Inventory key from the manifest
 * @returns {string} inventory path name
 */
function _localInventoryPath({ name, key }) {

```

[the remark this answers is not in the document]
Comment (victorlin) [reply resourceIndexer/inventory.js:307]: Oh, you're right! Sorry I misunderstood the first time. Fixed in #982

**Pull request merged**